### PR TITLE
pythonPackages.yappi: 1.2.5 -> 1.3.0

### PR DESCRIPTION
**Pull request by JamieMagee**
###### Motivation for this change

ZHF 20.09 #97479

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/yappi/default.nix [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+    rev = "30f94024a0e2e4fa21c220de6a0dc97b4cb2c319";
+    sha256 = "1kvwl3y3c2hivf9y2x1q1s8a2y724iwqd1krq6ryvsbg3inyh8qw";
   };
 
   patches = [ ./tests.patch ];
-
-  checkInputs = [ nose ];
+  checkInputs = [ gevent ];
+  checkPhase = ''
+    ${python}/bin/python run_tests.py
```
Comment (jonringer) [inline pkgs/development/python-modules/yappi/default.nix:18]: to support alternate python implementations
```suggestion
    ${python.interpreter} run_tests.py
```

**Comment by risicle** [comment]
Fails for me on macos 10.14

```
======================================================================
FAIL: test_filter_callback (test_functionality.BasicUsage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_functionality.py", line 198, in test_filter_callback
    self.assert_traces_almost_equal(r1, stats)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 0.16028 <= 0.343271 <= 0.28049 is not True. [tests/test_functionality.py:98 a      2      0.000000  0.200350  0.100175]

======================================================================
FAIL: test_tagging_walltime (test_tags.MultiThreadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_tags.py", line 50, in test_tagging_walltime
    self.assert_traces_almost_equal(t1, traces)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 1.6032471999999998 <= 3.277530999999999 <= 2.8056826 is not True. [..p/yappi/tests/utils.py:134 burn_io  20     0.000638  2.004059  0.100203]

======================================================================
FAIL: test_simple_tagging (test_tags.SingleThreadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_tags.py", line 169, in test_simple_tagging
    self.assert_traces_almost_equal(t1, traces)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 0.1665168 <= 0.33278699999999994 <= 0.2914044 is not True. [..p/yappi/tests/utils.py:134 burn_io  2      0.000000  0.208146  0.104073]

======================================================================
FAIL: test_basic_old_style (test_gevent.SingleThreadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_gevent.py", line 208, in test_basic_old_style
    self.assert_traces_almost_equal(r1, stats)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 1.2832392000000001 <= 2.327836 <= 2.2456686 is not True. [..p/yappi/tests/test_asyncio.py:43 a  2      0.000118  1.604049  0.802024]

======================================================================
FAIL: test_basic_old_style (test_asyncio.SingleThreadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_asyncio.py", line 65, in test_basic_old_style
    self.assert_traces_almost_equal(r1, stats)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 1.2832392000000001 <= 2.295179 <= 2.2456686 is not True. [..p/yappi/tests/test_asyncio.py:43 a  2      0.000118  1.604049  0.802024]

----------------------------------------------------------------------
Ran 97 tests in 41.206s

FAILED (failures=5, skipped=3)
```

This looks like another timing based test, which are notoriously flakey.

Wonder if we could find a sneaky way to disable the timing checks without disabling the surrounding tests.

**Comment by sumerc** [comment]
I will be trying to reproduce the error. However in the meantime: assert almost equal function accepts a err rate param which can be changed which I feel can be useful in your situation.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 101054 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python37Packages.yappi</li>    <li>python38Packages.yappi</li>  </ul></details>

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 101054 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python37Packages.yappi</li>    <li>python38Packages.yappi</li>  </ul></details>

**Pull request merged**